note: Tests appear to be borked as running `make test` on the `2.1.2` tag (as well as the latest `3.1.0` tag) results in a few fatal errors, such as:

```bash
=== RUN   TestStateTransitionsWithUpdateModules/Killed_in_ArtifactReboot
    state_test.go:4388: Killed in ArtifactReboot
panic: test timed out after 10m0s
```

I made sure that my changes didn't cause any additional tests to fail, but did not dig into the source of the above error

--- a/Documentation/update-modules-v3-file-api.md
+++ b/Documentation/update-modules-v3-file-api.md
@@ -111,6 +111,12 @@ The module should print one of the valid responses:
   reboot. **This is usually the best choice** for all modules that just require
   a normal reboot, but modules that reboot a peripheral device may need to use
   `Yes` instead, and implement their own method.
+* `AutomaticSkipStatusReport` - Almost identical semantics to `Automatic` - the
+  only difference being that **the mender client will not try to send a status
+  report to the server saying that the device is rebooting before carrying out
+  the reboot.** This response should be used specifically for update modules
+  which are expected to break upstream connectivity such that the client is
+  expected to be unable to report status to the server until after rebooting.
 * `Yes` - Mender will run the update module with the `ArtifactReboot`
   argument. Use this when you want to reboot a peripheral device that's
   connected to the host. Don't use this if you want to reboot the host that
@@ -126,6 +132,11 @@ performed after the `ArtifactReboot` state of all update modules that responded
 responded `Automatic` will always come after one that responded `Yes`, even
 though that may not be the original order in the Artifact.
 
+If any update module returns `AutomaticSkipStatusReport`, **the mender client
+will not try to report update status to the server before rebooting**, even if
+other modules responded with `Yes` or `Automatic`. The semantics are otherwise
+identical to `Automatic`.
+
 Unless all modules responded `No` in the `NeedsArtifactReboot` query, the
 `ArtifactReboot` state executes after `ArtifactInstall`. Inside this state it is
 permitted to call commands that reboot the system. However, if this happens,
@@ -140,7 +151,7 @@ cause problems, one of the following conditions should always be true:
   peripheral devices
 * All payloads reboot only peripheral devices, not the host system
 * If there is a mix, all payloads that want to reboot the host system respond
-  `Automatic` to the `NeedsArtifactReboot` query
+  `Automatic`/`AutomaticSkipStatusReport` to the `NeedsArtifactReboot` query
 
 If all update modules in the Artifact returned `No`, then the state scripts
 associated with this state, if any, will not run either.

--- a/datastore/statedata.go
+++ b/datastore/statedata.go
@@ -192,9 +192,10 @@ func (s *SupportsRollbackType) Set(value SupportsRollbackType) error {
 type RebootType string
 
 const (
-	RebootTypeNone      = ""
-	RebootTypeCustom    = "reboot-type-custom"
-	RebootTypeAutomatic = "reboot-type-automatic"
+	RebootTypeNone                      = ""
+	RebootTypeCustom                    = "reboot-type-custom"
+	RebootTypeAutomatic                 = "reboot-type-automatic"
+	RebootTypeAutomaticSkipStatusReport = "reboot-type-automatic-skip-status-report"
 )
 
 type RebootRequestedType []RebootType
@@ -205,7 +206,7 @@ func (r *RebootRequestedType) Get(n int) (RebootType, error) {
 			"Reboot information missing for payload %04d", n)
 	}
 	switch (*r)[n] {
-	case RebootTypeNone, RebootTypeCustom, RebootTypeAutomatic:
+	case RebootTypeNone, RebootTypeCustom, RebootTypeAutomatic, RebootTypeAutomaticSkipStatusReport:
 		return (*r)[n], nil
 	default:
 		return RebootTypeNone, errors.Errorf(

--- a/installer/installer.go
+++ b/installer/installer.go
@@ -76,6 +76,15 @@ const (
 	NoReboot = iota
 	RebootRequired
 	AutomaticReboot
+	// Reboot type intended to be used by an update module whose `ArtifactInstall` stage is
+	// expected to break upstream connectivity and therefore prohibit the mender client
+	// from notifying the server of a state transition into `ArtifactReboot`.
+	//
+	// Similar to `AutomaticReboot` in that this will cause the mender client to reboot
+	// the host device after handling all payloads. However, if ANY payload requests a reboot
+	// of this type, the mender client will NOT try to notify the server of the state transition
+	// to `ArtifactReboot` before rebooting the host device.
+	AutomaticRebootSkipStatusReport
 )
 
 var (

--- a/installer/modules.go
+++ b/installer/modules.go
@@ -775,6 +775,9 @@ func (mod *ModuleInstaller) NeedsReboot() (RebootAction, error) {
 	} else if output == "Automatic" {
 		log.Debug("Module needs host reboot")
 		return AutomaticReboot, nil
+	} else if output == "AutomaticSkipStatusReport" {
+		log.Debug("Module needs host reboot - skipping status report before rebooting")
+		return AutomaticRebootSkipStatusReport, nil
 	} else {
 		return NoReboot, fmt.Errorf("Unexpected reply from update module NeedsArtifactReboot query: %s",
 			output)

--- a/state.go
+++ b/state.go
@@ -881,7 +881,11 @@ func (is *UpdateInstallState) Handle(ctx *StateContext, c Controller) (State, bo
 		case datastore.RebootTypeNone:
 			// Do nothing.
 
-		case datastore.RebootTypeCustom, datastore.RebootTypeAutomatic:
+		case datastore.RebootTypeCustom:
+			fallthrough
+		case datastore.RebootTypeAutomatic:
+			fallthrough
+		case datastore.RebootTypeAutomaticSkipStatusReport:
 			// Go to reboot state if at least one payload requested it.
 			return NewUpdateRebootState(is.Update()), false
 
@@ -909,6 +913,8 @@ func (is *UpdateInstallState) handleRebootType(ctx *StateContext, c Controller) 
 			err = is.Update().RebootRequested.Set(n, datastore.RebootTypeCustom)
 		case installer.AutomaticReboot:
 			err = is.Update().RebootRequested.Set(n, datastore.RebootTypeAutomatic)
+		case installer.AutomaticRebootSkipStatusReport:
+			err = is.Update().RebootRequested.Set(n, datastore.RebootTypeAutomaticSkipStatusReport)
 		default:
 			state, cancelled := is.HandleError(ctx, c, NewFatalError(errors.New(
 				"Unknown reply from NeedsReboot. Should not happen")))
@@ -1428,9 +1434,31 @@ func (e *UpdateRebootState) Handle(ctx *StateContext, c Controller) (State, bool
 
 	log.Debug("handling reboot state")
 
-	merr := c.ReportUpdateStatus(e.Update(), client.StatusRebooting)
-	if merr != nil && merr.IsFatal() {
-		return NewUpdateRollbackState(e.Update()), false
+	// Avoid attempting to report status to the server if any payload indicated a reboot of
+	// type `RebootTypeAutomaticSkipStatusReport` (even if other modules exist which
+	// use other reboot types, we need to assume that upstream connectivity will be broken by
+	// the payload which indicated `RebootTypeAutomaticSkipStatusReport`).
+	shouldPhoneHomeBeforeReboot := true
+	for n, _ := range c.GetInstallers() {
+		rebootRequested, err := e.Update().RebootRequested.Get(n)
+		if err != nil {
+			return e.HandleError(ctx, c, NewTransientError(errors.Wrap(
+				err, "Unable to get requested reboot type")))
+		}
+		if rebootRequested == datastore.RebootTypeAutomaticSkipStatusReport {
+			log.Warnf(
+				"Found update with reboot type %s - NOT reporting status before reboot!",
+				datastore.RebootTypeAutomaticSkipStatusReport)
+
+			shouldPhoneHomeBeforeReboot = false
+		}
+	}
+
+	if shouldPhoneHomeBeforeReboot {
+		merr := c.ReportUpdateStatus(e.Update(), client.StatusRebooting)
+		if merr != nil && merr.IsFatal() {
+			return NewUpdateRollbackState(e.Update()), false
+		}
 	}
 
 	log.Info("rebooting device(s)")
@@ -1449,7 +1477,7 @@ func (e *UpdateRebootState) Handle(ctx *StateContext, c Controller) (State, bool
 				return NewUpdateRollbackState(e.Update()), false
 			}
 
-		case datastore.RebootTypeAutomatic:
+		case datastore.RebootTypeAutomatic, datastore.RebootTypeAutomaticSkipStatusReport:
 			systemRebootRequested = true
 		}
 	}
@@ -1549,7 +1577,11 @@ func (rs *UpdateRollbackState) Handle(ctx *StateContext, c Controller) (State, b
 		case datastore.RebootTypeNone:
 			// Do nothing.
 
-		case datastore.RebootTypeCustom, datastore.RebootTypeAutomatic:
+		case datastore.RebootTypeCustom:
+			fallthrough
+		case datastore.RebootTypeAutomatic:
+			fallthrough
+		case datastore.RebootTypeAutomaticSkipStatusReport:
 			// Enter rollback reboot state if at least one payload
 			// asked for it.
 			log.Debug("will try to rollback reboot the device")
@@ -1607,7 +1639,7 @@ func (rs *UpdateRollbackRebootState) Handle(ctx *StateContext, c Controller) (St
 				// VerifyRollbackReboot state regardless.
 			}
 
-		case datastore.RebootTypeAutomatic:
+		case datastore.RebootTypeAutomatic, datastore.RebootTypeAutomaticSkipStatusReport:
 			systemRebootRequested = true
 		}
 	}

--- a/state_test.go
+++ b/state_test.go
@@ -4840,3 +4840,54 @@ func TestAutomaticReboot(t *testing.T) {
 	require.NoError(t, err)
 	assert.Contains(t, string(logs), "exit status 99")
 }
+
+func TestAutomaticRebootSkipStatusReport(t *testing.T) {
+	tempDir, _ := ioutil.TempDir("", "logs")
+	defer os.RemoveAll(tempDir)
+
+	DeploymentLogger = NewDeploymentLogManager(tempDir)
+	defer func() {
+		DeploymentLogger.Disable()
+		DeploymentLogger = nil
+	}()
+
+	// This should not be necessary, but supposedly some other test is not
+	// cleaning up after itself.
+	log.Log = log.New()
+
+	log.AddHook(NewDeploymentLogHook(DeploymentLogger))
+	// We cannot remove hooks, so just clean up by resetting log.Log
+	// instead.
+	defer func() {
+		log.Log = log.New()
+	}()
+
+	ctx := &StateContext{
+		store:    store.NewMemStore(),
+		rebooter: system.NewSystemRebootCmd(stest.NewTestOSCalls("Called reboot", 101)),
+	}
+	u := &datastore.UpdateInfo{
+		Artifact: datastore.Artifact{
+			PayloadTypes: []string{"test-type"},
+		},
+		ID:              "abc",
+		RebootRequested: datastore.RebootRequestedType{datastore.RebootTypeAutomaticSkipStatusReport},
+	}
+	c := &stateTestController{}
+
+	// `.reportStatus` is set accordingly in `ReportUpdateStatus()` - if this field is never
+	// changed, that means we never called into `ReportUpdateStatus()`.
+	c.reportStatus = "dont-try-to-report-status-to-server"
+	rebootState := NewUpdateRebootState(u)
+
+	state, cancelled := rebootState.Handle(ctx, c)
+
+	assert.False(t, cancelled)
+	assert.IsType(t, &UpdateErrorState{}, state)
+
+	logs, err := DeploymentLogger.GetLogs("abc")
+	require.NoError(t, err)
+	assert.Contains(t, string(logs), "exit status 101")
+
+	assert.Equal(t, c.reportStatus, "dont-try-to-report-status-to-server")
+}

